cmem/encoder.go

```diff
@@ -0,0 +1,171 @@
+// Copyright ©2018 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package cmem
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math"
+	"reflect"
+	"unsafe"
+)
+
+var nativeEndian binary.ByteOrder
+
+func init() {
+	one := uint16(0x1)
+	nativeEndian = [...]binary.ByteOrder{
+		binary.BigEndian,
+		binary.LittleEndian,
+	}[(*[2]byte)(unsafe.Pointer(&one))[0]]
+}
+
+// Encoder is a wrapper type for information necessary to create and
+// subsequently write an in memory object to a PacketTable using Append().
+type Encoder struct {
+	Buf    []byte
```

Comment:
Why are these exported? If they need to be, there should be doc comments on them.

Reply:
On the contrary this is not the case for most of the fields in the lib :sunglasses: Will do though.

Reply:
Offset as well please.

Reply:
Is not exported any longer.

Reply:
Sorry, big-little dyslexia.

Reply:
:sunglasses: